src/main/scala/mesosphere/marathon/core/appinfo/EnrichedTask.scala

```diff
@@ -15,10 +15,27 @@ case class EnrichedTask(
     servicePorts: Seq[Int],
     reservation: Option[Reservation])
 
-object EnrichedTask {
-  def apply(instance: Instance, task: Task, healthCheckResults: Seq[Health],
-    servicePorts: Seq[Int] = Nil): EnrichedTask = {
-    new EnrichedTask(instance.runSpecId, task, instance.agentInfo, healthCheckResults, servicePorts = servicePorts,
-      reservation = instance.reservation)
+object EnrichedTasks {
+
+  object All {
+    def unapply(instance: Instance): Option[Iterable[EnrichedTask]] =
```

Comment:
Maybe just have a helper method that returns Iterable[EnrichedTask], and then match on that?